scripts/world/Level.gd

```diff
@@ -17,21 +17,20 @@ func _ready():
     connect("body_exited", self, "_on_area_exited")
 
 func _process(delta):
-    if active_level:
+    if is_active:
         var enter_level = Input.is_action_pressed("ui_select")
 
         if enter_level:
-            Game.current_level = active_level
-            get_tree().change_scene(active_level.scene)
+            Game.current_level_id = level_id
```

Comment:
Let's discuss this, but I believe we should restrict who can talk to the singleton (or at least write to it). I think this use-case can be served by having a signal `level_area_teleport_requested` (but with a better name 😸 ), with the world doing the actual work. The signal can pass the level id as a parameter.

Reply:
I was actually thinking about moving most of the functionality from Level to World, but kept it in as a future problem. I generally agree and can refactor it though. So you propose only World scene (out of all scenes in `world` folder) being able to talk to Game singleton?

Reply:
This is also now removed in #24 and it's being set from regular `Level` scene only.

Reply:
And I've created #28 to discuss this further.